### DISCUSSION TITLE
Nodes view

### DISCUSSION THREAD

**Discussion by Tanneguydv** [Tanneguydv/Pythonocc-nodes-for-Ryven#6] [Ideas]
I've seen that there is an unused function in Ryven : `NodesTreeListWidget.py`
[https://github.com/leon-thomm/Ryven/blob/master/Ryven/src/stuff/NodesTreeListWidget.py](https://github.com/leon-thomm/Ryven/blob/master/Ryven/src/stuff/NodesTreeListWidget.py)

It would be great if the user could have an overview of the functions available while programming, but I don't know how to implement it to Ryven correctly

**Comment by leon-thomm** [comment]
Indeed, that was part of a prototype for a widget listing all available nodes under their packages. Something like that I am probably going to add to the editor frontend in the next Ryven release.

**Reply by leon-thomm** [reply]
Here we go, coming for Ryven 3.1: extra nodes list widget on the left, also supports drag an drop into the scene.

![nodes-list](https://user-images.githubusercontent.com/29960068/143736004-d14e764b-4fbf-4516-8b8a-8c83d45839d7.png)

**Reply by Tanneguydv** [reply]
Great feature thanks!!
